docs/user/troubleshooting.md

```diff
@@ -145,3 +145,105 @@ Images:
 ```
 
 [kubernetes-debug]: https://kubernetes.io/docs/tasks/debug-application-cluster/
+
+### One or more nodes are never Ready (Network / CNI issues)
+
+You might see that one or more nodes are never ready, e.g
+
+```
+$ kubectl get nodes
+NAME                           STATUS     ROLES     AGE       VERSION
+ip-10-0-27-9.ec2.internal      NotReady   master    29m       v1.11.0+d4cacc0
+...
+```
+
+This usually means that, for whatever reason, networking is not available on the node. You can confirm this by looking at the detailed output of the node:
+
+```
+$ kubectl describe node ip-10-0-27-9.ec2.internal
+ ... (lots of output skipped)
+'runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni config uninitialized'
+```
+
+The first thing to determine is the status of the SDN. The SDN deploys three daemonsets:
+- *sdn-controller*, a control-plane component
+- *sdn*, the node-level networking daemon
+- *ovs*, the OpenVSwitch management daemon
+
+All 3 must be healthy (though only a single `sdn-controller` needs to be running). `sdn` and `ovs` must be running on every node, and DESIRED should equal AVAILABLE. On a healthy 2-node cluster you would see:
+
+```
+$ kubectl -n openshift-sdn get daemonsets
+NAME             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                     AGE
+ovs              2         2         2       2            2           beta.kubernetes.io/os=linux       2h
+sdn              2         2         2       2            2           beta.kubernetes.io/os=linux       2h
+sdn-controller   1         1         1       1            1           node-role.kubernetes.io/master=   2h
+```
+
+If, instead, you get a diferent error message:
+```
+$ kubectl -n openshift-sdn get daemonsets
+No resources found.
+```
+
+This means the network-operator didn't run. Skip ahead [to that section](#debugging-the-cluster-network-operator). Otherwise, let's debug the sdn.
+
+
+#### Debugging the openshift-sdn
+
+On the NotReady node, you need to find out which pods, if any are in a bad state. Be sure to substitute in the correct NodeName (or just remove it).
+```
+$ kubectl -n openshift-sdn get pod --field-selector "spec.nodeName=ip-10-0-27-9.ec2.internal"
+NAME                   READY   STATUS             RESTARTS   AGE
+ovs-dk8bh              1/1     Running            1          52m
+sdn-8nl47              1/1     CrashLoopBackoff   3          52m
+```
+
+Then, retrieve the logs for the SDN (and the OVS pod, if that is failed)
+```
+$ kubectl -n openshift-sdn logs sdn-8nl47
```

Comment:
nit: This block has a command with no output.  Personally I like:

    ```console

with a prompt tfor those, but this project hsd an existing patter of:

    ```sh

with no prompt.  More in #27.